test/integration/load_node_test.cpp

```diff
@@ -185,6 +185,20 @@ TEST(LoadNodeTest, DereferenceIteratorError) {
   EXPECT_THROW(node.begin()->begin()->Type(), InvalidNode);
 }
 
+TEST(NodeTest, MergeKeyScalarSupport) {
```

Comment:
Yes, I think multiple-level merges are legal, I'll add a test for them.

> Specifically, tests where it takes the main entry instead of the merge.

I'm not sure what you mean by this.